src/api-service/__app__/onefuzzlib/azure/creds.py

```diff
@@ -99,18 +103,77 @@ def get_regions() -> List[Region]:
     return sorted([Region(x.name) for x in locations])
 
 
-@cached
-def get_graph_client() -> GraphRbacManagementClient:
-    return GraphRbacManagementClient(get_msi(), get_subscription())
+class GraphQueryError(Exception):
+    def __init__(self, message: str, status_code: Optional[int]) -> None:
+        super(GraphQueryError, self).__init__(message)
+        self.message = message
+        self.status_code = status_code
+
+
+def query_microsoft_graph(
+    method: str,
+    resource: str,
+    params: Optional[Dict] = None,
+    body: Optional[Dict] = None,
+) -> Dict:
+    cred = get_identity()
+    access_token = cred.get_token(f"{GRAPH_RESOURCE}/.default")
+
+    url = urllib.parse.urljoin(f"{GRAPH_RESOURCE_ENDPOINT}/", resource)
+    headers = {
+        "Authorization": "Bearer %s" % access_token.token,
+        "Content-Type": "application/json",
+    }
+    response = requests.request(
+        method=method, url=url, headers=headers, params=params, json=body
+    )
+
+    if 200 <= response.status_code < 300:
+        if response.content and response.content.strip():
+            json = response.json()
+            if isinstance(json, Dict):
+                return json
+            else:
+                raise GraphQueryError(
+                    "invalid data expected a json object: HTTP"
+                    f" {response.status_code} - {json}",
+                    response.status_code,
+                )
+        else:
+            return {}
+    else:
+        error_text = str(response.content, encoding="utf-8", errors="backslashreplace")
+        raise GraphQueryError(
+            f"request did not succeed: HTTP {response.status_code} - {error_text}",
+            response.status_code,
+        )
+
+
+def query_microsoft_graph_list(
+    method: str,
+    resource: str,
+    params: Optional[Dict] = None,
+    body: Optional[Dict] = None,
+) -> List[Dict]:
+    result = query_microsoft_graph(
+        method,
+        resource,
+        params,
+        body,
+    )
+    value = result.get("value")
+    if isinstance(value, list):
+        return value
+    else:
+        raise GraphQueryError("Expected data containing a list of values", None)
 
 
 def is_member_of(group_id: str, member_id: str) -> bool:
-    client = get_graph_client()
-    return bool(
-        client.groups.is_member_of(
-            CheckGroupMembershipParameters(group_id=group_id, member_id=member_id)
-        ).value
+    body = {"groupIds": [group_id]}
+    response = query_microsoft_graph_list(
+        method="POST", resource=f"users/{member_id}/checkMemberGroups", body=body
     )
+    return group_id in response
```

Comment:
One last question: is this _ever_ true? It seems like it can't be, based on the types.

- `query_microsoft_graph_list()` returns `List[Dict]`
- `group_id` has type `str`

I'm surprised it even typechecks, so I may just be misreading something?

Reply:
The expected result is a [list of strings](https://docs.microsoft.com/en-us/graph/api/user-checkmembergroups?view=graph-rest-1.0&tabs=http#response-1)

I guess the type annotation is wrong in this particular case but the logic is correct